server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -332,6 +342,19 @@ private void searchLeaf(LeafReaderContext ctx, Weight weight, Collector collecto
                 } catch (CollectionTerminatedException e) {
                     // collection was terminated prematurely
                     // continue with the following leaf
+                } catch (QueryPhase.TimeExceededException e) {
+                    if (searchContext != null) {
+                        searchContext.setSearchTimedOut(true);
+                    } else {
+                        throw e;
+                    }
+                    return;
+                } catch (EarlyTerminatingCollector.EarlyTerminationException e) {
```

Comment:
It looks to me like adding support for `forceTermination` is pretty straightforward here: https://github.com/opensearch-project/OpenSearch/blob/1118dcf3727b6a2dee598159d65560adfc4d0dec/server/src/main/java/org/opensearch/search/query/EarlyTerminatingCollectorManager.java#L41

@reta do you have any more context or concerns with doing so?

Reply:
@jed326 if I remember correctly, with this change (in search leafs) it should work

Reply:
After looking into this some more and some discussion offline with @sohami I'm going to create a follow-up issue for this since it's actually not very trivial to implement. We will need to track the docs across all the threads in order to keep the same force termination behavior so we can have the discussion about how to do so over in the follow-up issue.